libqtile/widget/prompt.py

```diff
@@ -650,3 +661,20 @@ def cmd_info(self):
             text=self.text,
             active=self.active,
         )
+
+    def _dedup_history(self):
+        """Filter the history deque, clearing all duplicate values."""
+        self.history = {x: deque(copy.copy(self._dedup_deque(self.history[x])),
```

Comment:
Ummm... You got me xD. You're so right. The part of wrapping this into a deque is certainly redundant. I didn't notice that here I was already receiving a deque from `_dedup_deque`. I copied this from other part from my code (line 384 ~), where I was dealing with a list and is clear that I didn't thought enough about it. My bad :þ.

About the `copy.copy`, is the same reason. In this part I'm using a new deque, generated by `_dedup_deque` for every  completer, so there is no need to `copy.copy` it. In the other place, it was needed, because I was working with one list and... ummm, really in that place (line 384) I think I could simply use an slice of the entire list `[:]` and also get rid of the `copy` dependence. WTF was I thinking? >_<.

Well... This can wait, right? I mean, it has been waiting a year, so...
Two days ago (hence my delay answering you) my PC broke: Motherboard is gone. As soon as I can, I will fix this properly. I could edit it now, but I can't try it with qtile in my PC (because I don't have it working), pass the tests and all that stuff.